localization/twist_estimator/gyro_odometer/include/gyro_odometer/gyro_odometer_core.h

```diff
@@ -14,47 +14,46 @@
  * limitations under the License.
  */
 
-#pragma once
+#ifndef GYRO_ODOMETER_GYRO_ODOMETER_CORE_H_
 
-#include <ros/ros.h>
+#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <geometry_msgs/msg/twist_with_covariance_stamped.hpp>
+#include <sensor_msgs/msg/imu.hpp>
+#include <std_msgs/msg/float32.hpp>
 
 #include <tf2/transform_datatypes.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
 #include <tf2_ros/transform_listener.h>
 
-#include <std_msgs/Float32.h>
+#include <rclcpp/rclcpp.hpp>
 
-#include <geometry_msgs/TwistStamped.h>
-#include <geometry_msgs/TwistWithCovarianceStamped.h>
-#include <sensor_msgs/Imu.h>
 
-class GyroOdometer
+class GyroOdometer : public rclcpp::Node
 {
 public:
-  GyroOdometer(ros::NodeHandle nh, ros::NodeHandle private_nh);
+  GyroOdometer();
   ~GyroOdometer();
 
 private:
-  void callbackTwist(const geometry_msgs::TwistStamped::ConstPtr & twist_msg_ptr);
-  void callbackImu(const sensor_msgs::Imu::ConstPtr & imu_msg_ptr);
+  void callbackTwist(const geometry_msgs::msg::TwistStamped::ConstSharedPtr twist_msg_ptr);
+  void callbackImu(const sensor_msgs::msg::Imu::ConstSharedPtr imu_msg_ptr);
   bool getTransform(
     const std::string & target_frame, const std::string & source_frame,
-    const geometry_msgs::TransformStamped::Ptr & transform_stamped_ptr);
+    const geometry_msgs::msg::TransformStamped::SharedPtr transform_stamped_ptr);
 
-  ros::NodeHandle nh_;
-  ros::NodeHandle private_nh_;
+  rclcpp::Subscription<geometry_msgs::msg::TwistStamped>::SharedPtr vehicle_twist_sub_;
+  rclcpp::Subscription<sensor_msgs::msg::Imu>::SharedPtr imu_sub_;
 
-  ros::Subscriber vehicle_twist_sub_;
-  ros::Subscriber imu_sub_;
+  rclcpp::Publisher<geometry_msgs::msg::TwistStamped>::SharedPtr twist_pub_;
+  rclcpp::Publisher<geometry_msgs::msg::TwistWithCovarianceStamped>::SharedPtr twist_with_covariance_pub_;
+  // rclcpp::Publisher<std_msgs::Float32>::SharedPtr linear_x_pub_;
```

Comment:
Are the commented out publishers not needed anymore?

Reply:
Originally, these were not commented out however in the constructor, the publisher were never created (commented out) in the original implementation. I ported these regardless in case they are needed

Reply:
Ah, makes sense.